shell/platform/android/io/flutter/embedding/engine/plugins/util/GeneratedPluginRegister.java

```diff
@@ -14,16 +14,57 @@
   /**
    * Registers all plugins that an app lists in its pubspec.yaml.
    *
-   * <p>The Flutter tool generates a class called GeneratedPluginRegistrant, which includes the code
-   * necessary to register every plugin in the pubspec.yaml with a given {@code FlutterEngine}. The
-   * GeneratedPluginRegistrant must be generated per app, because each app uses different sets of
-   * plugins. Therefore, the Android embedding cannot place a compile-time dependency on this
-   * generated class. This method uses reflection to attempt to locate the generated file and then
-   * use it at runtime.
+   * <p>To let each plugin set itself up to listen to usage calls from Dart via platform channel,
+   * each plugin is given a chance to initialize itself and setup platform channel listeners in
+   * {@link
+   * io.flutter.embedding.engine.plugins.FlutterPlugin#onAttachedToEngine(io.flutter.embedding.engine.plugins.FlutterPlugin.FlutterPluginBinding)}.
+   *
+   * <p>The list of plugins needing setup is not known to the Flutter engine, but generated at build
+   * time based on the plugin dependencies specified in the Flutter project's pubspec.yaml.
+   *
+   * <p>The Flutter tool generates that list in a class called {@code GeneratedPluginRegistrant}.
+   * That class contains generated code to register every plugin in the pubspec.yaml with a {@link
+   * FlutterEngine}. That file is generated in the Flutter project's directory.
+   *
+   * <p>The {@link FlutterEngine} (when a {@link FlutterEngine} is explicitly created) or the {@link
```

Comment:
The parentheticals detract from the point of the sentence before it is established, it would be more clear if they were put at the end.

The GeneratedPluginRegistrant will automatically be called by the FlutterEngine (when a FlutterEngine is explicitly created) or the FlutterActivity (when a FlutterEngine is implicitly created).

Reply:
rephrased